src/System.Text.JsonLab/System/Text/Json/JsonReader.cs

```diff
@@ -83,6 +84,19 @@ public Utf8JsonReader(in ReadOnlySequence<byte> data)
             ValueType = JsonValueType.Unknown;
         }
 
+        public Utf8JsonReader(Stream stream)
+        {
+            _reader = default;
+            _isSingleSegment = true;
+            _buffer = ReadFromStreamAsync(stream).GetAwaiter().GetResult();
```

Comment:
Why call any async methods?

Reply:
I would put stream handling one or more levels higher, and handle the copy via Arraypool. Memorystream is not really light on array copy. If you still want to do stream copy, then check for canseek first, so you may be able to get the length of the stream to create a buffer with the correct size.

Reply:
I agree, it's strange to have the Stream API here.